app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -7,62 +7,64 @@
 
 import nl.tudelft.b_b_w.model.Block;
 import nl.tudelft.b_b_w.model.BlockFactory;
-import nl.tudelft.b_b_w.model.DatabaseHandler;
+import nl.tudelft.b_b_w.model.GetDatabaseHandler;
+import nl.tudelft.b_b_w.model.MutateDatabaseHandler;
 import nl.tudelft.b_b_w.model.TrustValues;
 
 /**
  * Performs the actions on the blockchain
  */
 
-public class BlockController {
-    /** Context of the block database */
+public class BlockController implements BlockControllerInterface {
+
+    /**
+     * Context of the block database
+     */
     private Context context;
 
-    /** Databasehandler to use */
-    private DatabaseHandler databaseHandler;
+    /**
+     *  Databasehandlers to use
+     */
+    private GetDatabaseHandler getDatabaseHandler;
+    private MutateDatabaseHandler mutateDatabaseHandler;
 
     /**
-     * contructor to initialize all the involved entities
+     * Constructor to initialize all the involved entities
      *
      * @param _context the instance
      */
     public BlockController(Context _context) {
         this.context = _context;
-        this.databaseHandler = new DatabaseHandler(this.context);
+        this.getDatabaseHandler = new GetDatabaseHandler(_context);
+        this.mutateDatabaseHandler = new MutateDatabaseHandler(_context);
     }
 
     /**
-     * Check if a block already exists in the database
-     * @param owner owner of the block
-     * @param key public key in the block
-     * @param revoked whether the block is revoked
-     * @return if the block already exists
+     * @inheritDoc
      */
+    @Override
     public final boolean blockExists(String owner, String key, boolean revoked) {
-        return databaseHandler.blockExists(owner, key, revoked);
+        return getDatabaseHandler.blockExists(owner, key, revoked);
     }
 
-
     /**
-     * adding a block to the blockchain
-     *
-     * @param block Block you want to add
-     * @return returns the block you added
+     * @inheritDoc
      */
+    @Override
     public final List<Block> addBlockToChain(Block block) {
         // Check if the block already exists
         String owner = block.getOwner();
-        Block latest = databaseHandler.getLatestBlock(owner);
+        Block latest = getDatabaseHandler.getLatestBlock(owner);
 
         if (latest == null) {
```

Comment:
Don't like these nested if else statements. Hard to test and bad practise. Maybe this can be split in more than one methods.